app/services/authentication_service.rb

```diff
@@ -1,25 +1,22 @@
 class AuthenticationService
 
-  def self.resolve(castle, user, session_id)
+  def self.resolve(castle, user, castle_authentication_id)
     authentication = Authentication.find_by(
```

Comment:
If we're cool with this code being run every time you reload the page then I guess we could simply fetch the Authentication object from the REST API instead of storing it locally. (Maybe that's what you're working on though)